### DISCUSSION TITLE
`pytest_configure` compatible with `--pyargs` ?

### DISCUSSION THREAD

**Discussion by skoudoro** [pytest-dev/pytest#12272] [General]
Hello pytest team,

it seems that `pytest_configure` is not compatible with `--pyargs`.  Can you confirm ?

here is my function in `conftest.py` (I confirm that `conftest.py` is in the root of the package):

```python
def pytest_configure(config):
    """Configure pytest options."""
    filterwarnings = [
        # FURY
        'ignore:.*You do not have FURY installed.*:UserWarning',
        # pkg_resources (via Ray)
        'ignore:Implementing implicit namespace packages.*:DeprecationWarning',
        'ignore:Deprecated call to `pkg_resources.*:DeprecationWarning',
        'ignore:pkg_resources is deprecated as an API.*:DeprecationWarning',
        ]
    for w_msg in filterwarnings:
        config.addinivalue_line("filterwarnings", w_msg)
```

my warnings are filtered when I use pytest from the source. However, it failed when using `pytest -svv --pyargs my_package`

is there an alternative ?  What should I use with `--pyargs` option ?

**Comment by bluetech** [comment]
A conftest inside of a `--pyargs` package is not considered an ["initial conftest"](https://docs.pytest.org/en/latest/how-to/writing_plugins.html#plugin-discovery-order-at-tool-startup) and `pytest_configure` only runs for initial conftests.

> is there an alternative ? What should I use with --pyargs option ?

You can put the conftest.py file in the directory from which you run your tests from or have your pytest config in, if that's an option.
